src/test/java/com/google/gcp_variant_transforms/library/BigQueryRowGeneratorTest.java

```diff
@@ -0,0 +1,140 @@
+    // and this row also contains record HQ=".,.", which should be [null,null] in the BQ row
+    TableRow firstRow = rows.get(0);
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.REFERENCE_NAME))
+            .isEqualTo("20");
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.START_POSITION))
+            .isEqualTo(14370);
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.END_POSITION))
+            .isEqualTo(14370);
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.NAMES))
+            .isEqualTo("rs6054257");
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.REFERENCE_BASES)).isEqualTo("G");
+
+    // AF field has type=`A` in VCFHeader, which should be moved from Info field to ALT field
+    List<TableRow> altMetadata = (List<TableRow>) firstRow.get(
+            VariantToBqUtils.ColumnKeyConstants.ALTERNATE_BASES);
+    TableRow altBase = altMetadata.get(0);
+    assertThat(altBase.get(VariantToBqUtils.ColumnKeyConstants.ALTERNATE_BASES_ALT)).isEqualTo("A");
+    assertThat(altBase.get("AF")).isEqualTo(0.5);
+
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.QUALITY)).isEqualTo(29);
+    Set<String> expectedFilter = new HashSet<>(Collections.singletonList("PASS"));
+    assertThat(firstRow.get(VariantToBqUtils.ColumnKeyConstants.FILTER)).
+            isEqualTo(expectedFilter);
+
+    assertThat(firstRow.get("NS")).isEqualTo(3);
+    assertThat(firstRow.get("DP")).isEqualTo(14);
+    assertThat(firstRow.get("DB")).isEqualTo(true);
+    assertThat(firstRow.get("H2")).isEqualTo(true);
+
+    List<TableRow> calls = (List<TableRow>) firstRow.get(VariantToBqUtils.ColumnKeyConstants.CALLS);
+    assertThat(calls.size()).isEqualTo(3);
+    // The third genotype contains HQ=".,.", test if the field value is [null, null]
+    TableRow rowWithMissingHQ = calls.get(2);
+    assertThat(rowWithMissingHQ.get(VariantToBqUtils.ColumnKeyConstants.CALLS_NAME))
+            .isEqualTo("NA00003");
+    assertThat(rowWithMissingHQ.get(VariantToBqUtils.ColumnKeyConstants.CALLS_PHASESET))
+            .isEqualTo("*");
+    assertThat(rowWithMissingHQ.get(VariantToBqUtils.ColumnKeyConstants.CALLS_GENOTYPE))
+            .isEqualTo(Arrays.asList(1, 1));
+    assertThat(rowWithMissingHQ.get("HQ")).isEqualTo(Arrays.asList(null, null));
+
+    // Test the fourth record in the test comment, there are some empty field in the record
+    TableRow fourthRow = rows.get(3);
+    assertThat(fourthRow.get(VariantToBqUtils.ColumnKeyConstants.NAMES)).isEqualTo(null);
+    // ALT field is ".", which should be null in the BQ row
+    List<TableRow> emptyAltMetadata = (List<TableRow>) fourthRow.get(
+            VariantToBqUtils.ColumnKeyConstants.ALTERNATE_BASES);
+    TableRow emptyAltBase = emptyAltMetadata.get(0);
+    assertThat(emptyAltBase.get(VariantToBqUtils.ColumnKeyConstants.ALTERNATE_BASES_ALT))
+            .isEqualTo(null);
```

Comment:
Are there any other rows we should test in addition to first and fourth row? Maybe some edge cases? 

I am not sure of any, just making sure we consider this.

Reply:
I have modified the VCF file to put all edge cases into one record row, which could make test more neat